### PR TITLE
WIP, ENH: bypass script 01

### LINKED ISSUE #321: ENH/API: script 01 obsolete for EEG
I am wondering if we can skip script 1; in my use case it simply copies/duplicates the EEG files (which I already duplicated by converting to bids ...). 

cc @hoechenberger @agramfort @sappelhoff

### PR DESCRIPTION
**Pull request by dengemann**
Closes #321.

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
let's see if it breaks any of the CIs.

**Comment by dengemann** [comment]
@hoechenberger @agramfort any ideas about the failing tests?

**Comment by hoechenberger** [comment]
Looking into this now

File: scripts/preprocessing/01-import_and_maxfilter.py [hunk 1]
```diff
@@ -466,10 +466,12 @@ def main():
     msg = 'Running Step 1: Data import'
```
Comment (hoechenberger) [inline scripts/preprocessing/01-import_and_maxfilter.py:466]:
```suggestion
    msg = 'Running Step 1: Maxwell filter
```

File: scripts/preprocessing/01-import_and_maxfilter.py [hunk 2]
```diff
@@ -466,10 +466,12 @@ def main():
...
-    parallel(run_func(subject, session) for subject, session in
-             itertools.product(config.get_subjects(), config.get_sessions()))
+    if config.use_maxwell_filter:
+        parallel, run_func, _ = parallel_func(run_maxwell_filter,
+                                              n_jobs=config.N_JOBS)
+        parallel(run_func(subject, session) for subject, session in
+                 itertools.product(config.get_subjects(),
+                                   config.get_sessions()))
 
     msg = 'Completed Step 1: Data import'
```
Comment (hoechenberger) [inline scripts/preprocessing/01-import_and_maxfilter.py:476]:
```suggestion
    msg = 'Completed Step 1: Maxwell filter
```

File: scripts/preprocessing/02-frequency_filter.py [hunk 3]
```diff
@@ -72,7 +72,15 @@ def run_filter(subject, run=None, session=None):
     logger.info(gen_log_message(message=msg, step=2, subject=subject,
                                 session=session, run=run,))
 
-    raw = mne.io.read_raw_fif(raw_fname_in)
+    read_raw = mne.io.read_raw_fif
```
Comment (hoechenberger) [inline scripts/preprocessing/02-frequency_filter.py:75]: Hm, we do this nowhere else in the pipeline, do you feel strongly about keeping this?

**Review by hoechenberger — COMMENTED** [review]
@dengemann I believe I found the issue, will push an attempted fix shortly.

**Comment by hoechenberger** [comment]
Fixed the issue with the "missing" events, but now there's something else showing up. Digging into this…

**Comment by hoechenberger** [comment]
Well. This blew up.

**Comment by dengemann** [comment]
@hoechenberger I think we're moving away from the purpose of the pull request which was to skip any script that runs before the 02 frequency filtering.

**Comment by hoechenberger** [comment]
We're skipping unless the user wants to add or remove channels or rename events

Maybe we should have a call, as this thing is much more complex than at first anticipated. We're doing many more things in the import step than just copying the data.

**Comment by dengemann** [comment]
> Maybe we should have a call, as this thing is much more complex than at first anticipated. We're doing many more things in the import step than just copying the data.

Yes, but even these things are not mandatory for all datasets and situations, no?
Sure let's have a call.

**Comment by agramfort** [comment]
discusses with @hoechenberger 

the plan is to add an `import_raw` function in config.py and use this function in the first script we need.

**Comment by hoechenberger** [comment]
Will make a PR later today!

**Comment by hoechenberger** [comment]
Ok I think I've decluttered most of the functionality and refactored it into numerous small functions. Now it's time to re-assemble things and eventually get rid of script 00 (import)

**Pull request closed without merging**